core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/InputGuardrailResult.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InputGuardrailResult {
+
+    private static final InputGuardrailResult SUCCESS = new InputGuardrailResult();
+
+    private final boolean success;
+    private List<Failure> failures;
+
+    private InputGuardrailResult() {
+        this.success = true;
+    }
+
+    private InputGuardrailResult(List<Failure> failures) {
+        this.success = false;
+        this.failures = failures;
+    }
+
+    public static InputGuardrailResult success() {
+        return SUCCESS;
+    }
+
+    public static InputGuardrailResult fail(String message) {
+        return new InputGuardrailResult(List.of(new Failure(message)));
+    }
+
+    public static InputGuardrailResult fail(String message, Throwable cause) {
```

Comment:
Each validation failure has a message and optionally also an exception that can be used to track the original cause.